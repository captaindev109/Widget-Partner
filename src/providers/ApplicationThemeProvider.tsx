import { ThemeProvider, DefaultTheme } from 'styled-components';
import GlobalStyles from '@/styles/Global';
import React from 'react';
import { ChildrenReturnType } from '@/types/types';

interface ApplicationThemeProviderProps {
  children?: ChildrenReturnType;
  theme: DefaultTheme;
}

const ApplicationThemeProvider = ({ children, theme }: ApplicationThemeProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};

export default ApplicationThemeProvider;
