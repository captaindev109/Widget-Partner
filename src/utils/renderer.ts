import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
