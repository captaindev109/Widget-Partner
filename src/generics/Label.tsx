import { device } from '@/utils/constants';
import { FC, ReactNode } from 'react';
import styled from 'styled-components';

export interface LabelProps {
  children?: ReactNode;
  htmlFor?: string;
}

const StyledText = styled.label<LabelProps>`
  font-size: ${(props) => props.theme.size.fontSizes.f2};
  line-height: ${(props) => props.theme.size.lineHeights.lh2};
  letter-spacing: ${(props) => props.theme.size.letterSpacings.ls1};
  font-family: ${(props) => props.theme.fonts.FAVORIT};
  @media ${device.mobileS} {
    ${(props) =>
      `
      font-size: ${props.theme.size.fontSizes.f2};
      line-height: ${props.theme.size.lineHeights.lh2};
      letter-spacing: ${props.theme.size.letterSpacings.ls1};
       `}
  }
  @media ${device.laptop} {
    ${(props) =>
      `
      font-size: ${props.theme.size.fontSizes.f2};
      line-height: ${props.theme.size.lineHeights.lh2};
      letter-spacing: ${props.theme.size.letterSpacings.ls1};
       `}
  }
  text-overflow: ellipsis;
  overflow: hidden;
`;

const Label: FC<LabelProps> = ({ children, htmlFor, ...newProps }) => {
  return (
    <StyledText htmlFor={htmlFor} {...newProps}>
      {children}
    </StyledText>
  );
};
export default Label;
