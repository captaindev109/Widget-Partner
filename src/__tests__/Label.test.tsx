import { render, RenderResult } from '@testing-library/react';
import queries from '@testing-library/dom/types/queries';
import renderer from 'react-test-renderer';
import ApplicationThemeProvider from '@/providers/ApplicationThemeProvider';
import defaultTheme from '@/theme/theme';
import Label from '@/generics/Label';

const children = 'We offer insurance';

describe('Test cases of the Label component', () => {
  let renderedComponent: RenderResult<typeof queries, HTMLElement>;

  beforeEach(() => {
    renderedComponent = render(
      <ApplicationThemeProvider theme={defaultTheme}>
        <Label>{children}</Label>
      </ApplicationThemeProvider>
    );
  });

  test('to see if the component render without crashing', () => {
    render(
      <ApplicationThemeProvider theme={defaultTheme}>
        <Label />
      </ApplicationThemeProvider>
    );
  });

  test('to see if the Label children is correct', () => {
    const { getByText } = renderedComponent;
    expect(getByText(children)).toBeTruthy();
  });

  it('Snapshot test should pass', () => {
    const snap = renderer
      .create(
        <ApplicationThemeProvider theme={defaultTheme}>
          <Label />
        </ApplicationThemeProvider>
      )
      .toJSON();
    expect(snap).toMatchSnapshot();
  });
});
