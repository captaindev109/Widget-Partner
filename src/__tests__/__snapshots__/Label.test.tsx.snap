// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases of the Label component Snapshot test should pass 1`] = `
<label
  className="Label__StyledText-sc-76lssq-0 eBRjDt"
/>
`;
